RecoTracker/TransientTrackingRecHit/python/TTRHBuilderWithTemplate_cfi.py

```diff
@@ -11,7 +11,9 @@
 from Configuration.Eras.Modifier_trackingPhase2PU140_cff import trackingPhase2PU140
 trackingPhase2PU140.toModify(TTRHBuilderAngleAndTemplate, Phase2StripCPE = cms.string('Phase2StripCPE'))
 
-# uncomment these two lines to turn on Cluster Repair CPE
-from Configuration.Eras.Modifier_phase1Pixel_cff import phase1Pixel
-phase1Pixel.toModify(TTRHBuilderAngleAndTemplate, PixelCPE = cms.string('PixelCPEClusterRepair'))
+#Turn on cluster repair if not running at 0T
+from Configuration.StandardSequences.MagneticField_cff import ParabolicParametrizedMagneticFieldProducer
+if ParabolicParametrizedMagneticFieldProducer.valueOverride != 0:
```

Comment:
Mostly out of curiosity, do I guess correctly that this trick works because
- for 0T jobs the generated configuration loads `MagneticField_0T_cff` instead of `MagneticField_cff`, and
- the generated configuration loads magnetic field cff before `Reconstruction_cff`

?

Reply:
which jobs would these be?
```git grep MagneticField_0T_cff | grep -v /test``` doesn't show anything production-like.

Reply:
Yes this was my idea behind the fix. 

There aren't any runTheMatrix tests with 0T (which is why this issue was not noticed before).
So I was testing with a config @mmusich provided : /afs/cern.ch/user/m/musich/public/forCristina/step3_RAW2DIGI_L1Reco_RECO_RECOSIM_EI_PAT_VALIDATION_DQM.py

I think it is standard to load the magnetic field before the reco, but I don't think its required right? So maybe there is some safer way to do this

Reply:
I'm not sure what is the connection to the T0 replay.
Which job at T0 is this expected to represent?
Express or Prompt do not load MagneticField_0T_cff; the default is AutoFromDBCurrent

Reply:
@slava77,
that configuration pointed by Oz is not supposed to represent *any* job at Tier0.
It is actually a TTBar18 workflow modified to run without magnetic field. I stumbled in the same issue as the Tier0 replay when studying 0T reconstruction for totally orthogonal reasons, and notified the 2D template reco developers, before the replayed happened at all. 
Now the fix from Oz solves my problem, but unfortunately doesn't work in data as, as you point out, the magnetic field map is loaded according to the magnet current stored in the conditions.
@OzAmram, @cmantill I think the way to fix software-wise, if a calibration for 0T cannot be provided, is to modify the `ESProducer` not to look for labels which are not the default one **and** modify the C++ 2D template code not to run if the magnet current is not the nominal one.

Reply:
I agree that the fix should be done in some other way, somewhere on the ES producer side.

Reply:
Ok sorry for the confusion about the Tier 0.
Is there a test case any can provide then, what would represent what the Tier 0 would run? 

Is there any other way to check the magnetic field value in the event configuration then? 
I was hoping to avoid modifying the ESProducer and CPE to have a workaround in case of a different magnetic field, but if that is the only way it can be done.

Reply:
> Is there a test case any can provide then, what would represent what the Tier 0 would run?

Chris provided a nice recipe to reproduce the Tier-0 replay here:  https://hypernews.cern.ch/HyperNews/CMS/get/tier0-Ops/2129/1/1/1/1/1/1.html

Reply:
@mmusich ok I pushed my commits in this branch: https://github.com/cmantill/cmssw/tree/cpe-at-0T. Oz will cherry pick them up soon. I have tested this locally using both your recipe [1] and Chris's recipe (without modifying CPE to templateReco) [2].

[1] /afs/cern.ch/user/m/musich/public/forCristina/step3_RAW2DIGI_L1Reco_RECO_RECOSIM_EI_PAT_VALIDATION_DQM.py
[2] 
cmsrel CMSSW_10_6_0
cd CMSSW_10_6_0/src
cmsenv
scram b -j 4
cp /afs/cern.ch/work/v/vjankaus/public/20190523/Express-014a1f93-4f18-4b1c-9535-065483b74c94-2-logArchive.tar.gz  .
tar xf Express-014a1f93-4f18-4b1c-9535-065483b74c94-2-logArchive.tar.gz
cd job/WMTaskSpace/cmsRun1
edmConfigDump PSet.py > dump.py
sed -i 's|106X_dataRun3_Express_Candidate_2019_05_22_20_52_06|106X_dataRun3_Express_Candidate_2019_05_24_21_36_45|g' dump.py
cmsRun dump.py